app/src/main/java/com/example/learingrealmandretrofit/fragment/CardFragment.kt

```diff
@@ -38,38 +37,30 @@ class CardFragment : Fragment() {
         view: View,
         savedInstanceState: Bundle?
     ) {
-        Realm.init(context)
-
         deleteAllCardsRealm()
         getAllCardRetrofit()
 
-        binding.floatingActionButtonCard.setOnClickListener {
+        binding.buttonCreateCard.setOnClickListener {
             findNavController().navigate(R.id.action_cardFragment_to_dialogCreateOrChangeCard)
         }
 
-        findNavController().currentBackStackEntry?.savedStateHandle?.getLiveData<String>(
-            DialogDeleteCard.noDeleteWordKey
-        )?.observe(
-            viewLifecycleOwner
-        ) {
-            binding.recyclerCard.adapter?.notifyItemChanged(currentSwipePosition)
-        }
-
         val item = object : SwipeToDeleteCard(requireContext()) {
             override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {
-                currentSwipePosition = viewHolder.absoluteAdapterPosition
                 val currentSwipeCardId = pullCardsRealm()[viewHolder.absoluteAdapterPosition]?.id
-                findNavController().navigate(
-                    R.id.action_cardFragment_to_dialogDeleteCard,
-                    bundleOf(DialogDeleteCard.deleteWordKey to currentSwipeCardId))
+                if (currentSwipeCardId != null) {
+                    val arrow = CardFragmentDirections.actionCardFragmentToDialogDeleteCard(currentSwipeCardId)
+                    findNavController().navigate(arrow)
+                    binding.recyclerCard.adapter?.notifyItemChanged(viewHolder.absoluteAdapterPosition)
+                }
             }
         }
         val itemTouchHelper = ItemTouchHelper(item)
         itemTouchHelper.attachToRecyclerView(binding.recyclerCard)
     }
 
    private fun getAllCardRetrofit() {
-       val retrofitData = BaseApi.retrofit.getCards()
+       val token = context?.user()?.token ?: ""
+       val retrofitData = BaseApi.retrofit.getCards(token)
```

Comment:
1. If token is not present - this request should not be made.
2. You should not fetch and insert token on each authenticated API call.